models/quantity.rb

```diff
@@ -19,8 +19,8 @@
 
 module Models
   class Quantity < Base
-    field :fill, default: false
-    field :empty, default: false
+    field :fill # Active_hash did not record default values set to false
+    field :empty # because the condition to reach add_default_value is `if options[:default]`
```

Comment:
But then we always need to remember to make `!!` with fill and empty -- and I don't really like that :/ there will be bugs. In any case, tell me what you think about (i) having the defaults in create or (ii) fixing the active_hash bug with `options.has_key?[:default]`

Reply:
There is only an issue with our protobuf because we expect boolean values.

Anyway, as we use our own fork for active_hash, we indeed can change the condition to allow the value `false`.
As this is the default value, we will in any case never return it with as_json.

Moreover `!!` is often used in Mapotempo Web to turn nil values into booleans